### DISCUSSION TITLE
[New] Proposition d'ajout de pratique : « Limiter l'utilisation de canvas / webgl »

### DISCUSSION THREAD

**Discussion by yannkozon** [cnumr/best-practices#100] [Bonnes pratiques] [answered]
A l'instar de Flash à une certaine époque de nombreux utilisent les canvas et webgl pour intégrer des contenus et animations sur les sites. Le besoin en ressource machine est très important.

**Comment by ACTLEM** [comment]
A voir si cela ne pourrait pas être intégré dans la modification de la BP 12, voir la discussion https://github.com/cnumr/best-practices/discussions/28?

**Reply by ACTLEM** [reply]
Néanmoins comme le dit @fzaninotto, le webgl utilise le GPU et permet de moins solliciter le CPU, ce qui a pour effet d'avoir un ressenti utilisateur plus fluide et donc moins de risque d'obsolescence perçue. Qui plus est Webgl est "intégré" dans tous les navigateurs et est bas niveau car s'appuie sur OpenGl.
Après, la nécessité d'avoir des animations (souvent 3D) dans un service web est assez restreinte. Souvent cela touche le domaine du jeu vidéo. Il y a parfois des utilisations excessives de Webgl pour faire "fun", mais j'ai tendance à dire qu'à ce moment là cette BP serait un cas particulier/exemple à ne pas faire d'une des BP "Eliminer les fonctionnalités inutiles", "Quantifier le besoin" ...

**Comment by fzaninotto** [comment]
Je me permets de challenger cette proposition: utiliser WebGL permet de se based sur le GPU, mieux optimisé en terme de consommation d'énergie pour des calculs très parallèles que le CPU. Dans le cas d'une page avec de nombreux graphiques par exemple, utiliser WebGL au lieu de SVG ou pire de HTML peut représenter un gain très conséquent.

Pour trancher, il faudrait des implémentations d'une même page avec et sans WebGL, avec une mesure comparée de la consommation d'énergie. A moins d'avoir cette preuve, cette pratique reste pour moi une croyance.

**Reply by yannkozon** [reply]
L’utilisation de canvas et/ou webgl a son utilité, d’où ma formulaire de limiter ces usages. On se retrouve malheureusement aujourd’hui avec des sites qui utilisent ces fonctionnalités pour des sites entiers qui a un impact fort sur les machines anciennes (mais aussi les récentes) et donnent souvent des sites inaccessibles pour certains public.

**Comment by ACTLEM** [comment, marked as answer]
Note équipe projet: Ok pour la création d'une bonne pratique sur les Canvas. Sur la notion de webGL, il y a débat par contre il nous semble opportun de reprendre l'argumentaire de la discussion https://github.com/cnumr/best-practices/discussions/28#discussioncomment-1528417 (note la suppression ou non de la BP 12 n'est pas encore actée).

Issue créée: https://github.com/cnumr/best-practices/issues/182
